### DISCUSSION TITLE
POST_FLAT_FILE_LISTINGS_DATA Documentation

### DISCUSSION THREAD

**Discussion by DmytroShumeyko** [amzn/selling-partner-api-models#3947] [General] [answered]
Hello! I can't find documentation about POST_FLAT_FILE_LISTINGS_DATA details:

1. I'm looking for a description of these attributes: https://developer-docs.amazon.com/sp-api/docs/listings-feeds-attributes#post_flat_file_listings_data
2. I'm looking for documentation on which fields are required to add products directly to a merchant's FBA inventory. If it was possible using the same MWS report, I believe I can do the same using this one.

Thanks for your help.

**Comment by shreeharsh-a** [comment]
@MeenaAmz

**Comment by chapmanjw** [comment, marked as answer]
@DmytroShumeyko That feed type via the Feeds API is being deprecated. Here is the announcement for that: https://developer-docs.amazon.com/sp-api/changelog/deprecation-of-feeds-api-support-for-xml-and-flat-file-listings-feeds. 

All programmatic submissions for listings should go through the Listings API directly or use the JSON_LISTINGS_FEED feed type with the Feeds API (the payloads are interchangeable between the two). I would recommend starting here: https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide. 

You will essentially use the Product Type Definitions API to get JSON Schemas defining the valid payloads for each product type (i.e., which attributes are required, etc.). And then use either the Listings API or JSON_LISTINGS_FEED - Feeds API to submit the data.
